x-pack/plugin/core/src/main/resources/logs-default-pipeline.json

```
@@ -0,0 +1,24 @@
+{
+  "processors": [
+    {
+      "set": {
+        "description": "If '@timestamp' is missing, set it with the ingest timestamp",
+        "field": "@timestamp",
+        "override": false,
+        "value": "{{{_ingest.timestamp}}}"
```

This prevents having to evaluate a mustache template every time which is quite expensive
```suggestion
        "copy_from": "_ingest.timestamp"
```

---

I am not sure it is possible though. I tried it and the test fails. 
IIUC, the ingest metadata fields are not actual fields, they are only available in the response metadata somehow. I will take another look

---

Hm, this worked for me in my POC: https://gist.github.com/felixbarny/a9a2f6243153d5508643fd95ac968a88#file-routing-yml-L130-L134

---

@jbaiera can you help me understand that?
The error when changing to @felixbarny's suggestion is:
```
error:
  root_cause:
  - type: "document_parsing_exception"
    reason: "[4:15] failed to parse field [@timestamp] of type [date] in document\
      \ with id 'EUujBIgBUxyWBQiNfsUS'. Preview of field's value: '_ingest.timestamp'"
    stack_trace: "org.elasticsearch.index.mapper.DocumentParsingException: [4:15]\
      \ failed to parse field [@timestamp] of type [date] in document with id 'EUujBIgBUxyWBQiNfsUS'.\
      \ Preview of field's value: '_ingest.timestamp'\n\tat org.elasticsearch.server@8.9.0-SNAPSHOT/org.elasticsearch.index.mapper.FieldMapper.rethrowAsDocumentParsingException(FieldMapper.java:215)\n\
      \tat org.elasticsearch.server@8.9.0-SNAPSHOT/org.elasticsearch.index.mapper.FieldMapper.parse(FieldMapper.java:168)\n\
      \tat org.elasticsearch.server@8.9.0-SNAPSHOT/org.elasticsearch.index.mapper.DocumentParser.parseObjectOrField(DocumentParser.java:379)\n\
      \tat org.elasticsearch.server@8.9.0-SNAPSHOT/org.elasticsearch.index.mapper.DocumentParser.parseValue(DocumentParser.java:585)\n\
...
```
I believe `Preview of field's value: '_ingest.timestamp'` means that there is no such field.
If that is correct, can you explain why we cannot get this field directly this way?

---

Did you set `"value": "_ingest.timestamp"` instead of `"copy_from": "_ingest.timestamp"`?

---

> Did you set "value": "_ingest.timestamp" instead of "copy_from": "_ingest.timestamp"?

Yes, I did 🤦‍♂️